### PR TITLE
Update dependencies to fix CVE-2022-25647

### PR DESCRIPTION
**Pull request by Torch3333**
This fixes [CVE-2022-25647](https://github.com/advisories/GHSA-4jrv-ppp4-jm57 ) targeting `com.google.code.gson:gson`. 
As a result, multiple dependencies needed to be updated such as grpc so the proto files needed to be regenerated.
 
Besides, this raised new spotbugs warning as a side effects. They are addressed in https://github.com/scalar-labs/scalardb/pull/589/commits/9150618116936154167ceb7551260e8d093c4985 .

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -32,9 +32,9 @@ subprojects {
         postgresqlDriverVersion = '42.3.3'
         oracleDriverVersion = '19.8.0.0'
         sqlserverDriverVersion = '8.4.1.jre8'
-        grpcVersion = '1.38.0'
-        protocVersion = '3.17.2'
-        protobufVersion = '3.17.2'
+        grpcVersion = '1.46.0'
+        protocVersion = '3.20.1'
+        protobufVersion = '3.20.1'
```
Comment (brfrn169) [inline build.gradle:37]: These upgradings are backward compatible?

**Review by brfrn169 — COMMENTED** [review]
I left a question. Please take a look when you have time!

File: build.gradle (hunk 1 above)
Comment (Torch3333) [reply build.gradle:37]: Yes, I think so. To confirm it, I ran the Scalar DB server integration test with the following configuration :
 - `server:integrationTestScalarDbServer`(`master` branch version) targeting Scalar DB Server (`fix_CVE-2022-25647` branch version) running on Mysql
  - `server:integrationTestScalarDbServer`(`fix_CVE-2022-25647` branch version) targeting Scalar DB Server (`master` branch version) running on Mysql

Comment (brfrn169) [reply build.gradle:37]: Thank you for the confirmation!

**Review by brfrn169 — APPROVED** [review]
LGTM, thank you!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**